pylint/checkers/logging.py

```diff
@@ -232,6 +232,23 @@ def _check_format_string(self, node, format_arg):
             self.add_message('logging-too-few-args', node=node)
 
 
+def is_complex_format_str(node):
+    """ Checks if node represents a string with complex formatting specs.
+
+    Args:
+        node (astroid.node_classes.NodeNG): AST node to check
+    Returns:
+        bool: True if inferred string uses complex formatting, False otherwise
+    """
+    inferred = utils.safe_infer(node)
+    if inferred is None or not isinstance(inferred.value, six.string_types):
+        return True
+    for _, _, format_spec, _ in string.Formatter().parse(inferred.value):
+        if format_spec:
```

Comment:
maybe some simple formatter specs may be detected and converted to suggestions (e.g. `!r` -> `%r`)